test/server/http/admin_filter_test.cc

```diff
@@ -0,0 +1,79 @@
+#include "server/http/admin_filter.h"
+
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::InSequence;
+using testing::NiceMock;
+
+namespace Envoy {
+namespace Server {
+
+class AdminFilterTest : public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  AdminFilterTest() : filter_(adminServerCallback), request_headers_{{":path", "/"}} {
+    filter_.setDecoderFilterCallbacks(callbacks_);
+  }
+
+  NiceMock<MockInstance> server_;
+  Stats::IsolatedStoreImpl listener_scope_;
+  AdminFilter filter_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  Http::TestRequestHeaderMapImpl request_headers_;
+
+  static Http::Code adminServerCallback(absl::string_view path_and_query,
+                                        Http::ResponseHeaderMap& response_headers,
+                                        Buffer::OwnedImpl& response, AdminFilter& filter) {
+    // silence compiler warnings for unused params
+    (void)path_and_query;
```

Comment:
sorry; one more nit. Use UNREFERENCED_PARAMETER from source/common/common/macros.h here rather than the explicit void cast.